lib/ExpandingTextArea/index.js

```diff
@@ -0,0 +1,105 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+class ExpandingTextArea extends Component {
+  static propTypes = {
+    placeholder: PropTypes.string.isRequired,
+    handleOnChange: PropTypes.func,
+    value: PropTypes.string,
+    focusOnMount: PropTypes.bool,
+    setValue: PropTypes.bool
+  };
+
+  static defaultProps = {
+    handleOnChange: () => {},
+    value: '',
+    focusOnMount: false,
+    setValue: false
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = { inputValue: '', rows: 1, rowSize: 0, padding: 0 };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  componentDidMount() {
+    this.setInitialRows();
+
+    if (this.props.focusOnMount) {
+      this.input.focus();
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.value !== nextProps.value) {
+      if (!nextProps.value) {
+        return this.setState({ rows: 1 });
+      }
+      return this.resizeTextArea();
+    }
+    return null;
+  }
+
+  setInitialRows() {
+    const rowSize = parseInt(
+      window.getComputedStyle(this.input).lineHeight,
```

Comment:
Might be worth assigning this in a const above? e.g

```
const inputStyles = window.getComputedStyle(this.input);
```

Makes this method a little DRYer.